src/error.rs

```diff
@@ -5,12 +5,15 @@ use std::{error, fmt, io};
 pub enum Error {
     /// Errors with reading or writing to IO.
     Io(io::Error),
+    /// TODO: Is this the right name for this error.
+    InputEmpty,
```

Comment:
@mossberg Thoughts on this? The result of iterating over a reader is weird. See the implementation of the `,` instruction for more.